### PR TITLE
feat: Aplib-Unity integration

### PR DESCRIPTION
**Pull request by joachimdekker**
## Description

This PR proposes to add the testrunner for testing frameworks such as Unity Test Framework to use.

## Testability

The testrunner literally tests the game, so testing is a bit hard. You could try to run this version in Unity, but installing that is quite the hassle.

## Checklist
- [ ] Set the proper pull request name
- [ ] Merged main into your branch

### CODE REVIEW

File: Aplib.Integrations.Unity.Tests/TestRunnerTests.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+{
+    [Fact]
+    public void Test_ShouldCallAgentUpdate_WhenAgentStatusIsUnfinished()
+    {
+        // Arrange
+        Mock<IAgent> mockAgent = new();
+        mockAgent.SetupSequence(agent => agent.Status)
+            .Returns(CompletionStatus.Unfinished)
+            .Returns(CompletionStatus.Success);
+        mockAgent.Setup(agent => agent.Update());
```
Comment (JensSteenmetz) [inline Aplib.Integrations.Unity.Tests/TestRunnerTests.cs:17]: Not necessary.
```suggestion
```

File: Aplib.Integrations.Unity.Tests/TestRunnerTests.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
+using Aplib.Core;
+using Moq;
+using System.Collections;
+
+namespace Aplib.Integrations.Unity.Tests;
+
+public class UnitTest1
```
Comment (JensSteenmetz) [inline Aplib.Integrations.Unity.Tests/TestRunnerTests.cs:7]:
```suggestion
public class TestRunnerTests
```

File: Aplib.Integrations.Unity.Tests/TestRunnerTests.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+{
+    [Fact]
+    public void Test_ShouldCallAgentUpdate_WhenAgentStatusIsUnfinished()
```
Comment (JensSteenmetz) [inline Aplib.Integrations.Unity.Tests/TestRunnerTests.cs:10]:
```suggestion
    public void Test_WhenAgentStatusIsUnfinished_ShouldCallAgentUpdate()
```

File: Aplib.Integrations.Unity/TestRunner.cs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        /// <returns>An IEnumerator that can be used to control the execution of the test.</returns>
+        public IEnumerator Test()
+        {
+            while (_agent.Status == CompletionStatus.Unfinished)
+            {
+                // Perform computation or update the agent here
+                _agent.Update();
+
+                // Wait for the next frame
+                yield return null;
```
Comment (JensSteenmetz) [inline Aplib.Integrations.Unity/TestRunner.cs:35]: Comment conventions are to end comments with a period, [see Microsoft Docs](https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/coding-conventions#comment-style).
```suggestion
                // Perform computation or update the agent here.
                _agent.Update();

                // Wait for the next frame.
                yield return null;
```

**Review by JensSteenmetz — CHANGES_REQUESTED** [review]
Some small suggestions, but looks great overall!!

File: Aplib.Integrations.Unity/TestRunner.cs (hunk 4 above)
Comment (SilasPeters) [reply Aplib.Integrations.Unity/TestRunner.cs:35]: Damn if this ain't nitpicking.

+1 though

File: Aplib.Integrations.Unity/TestRunner.cs [hunk 5]
```diff
@@ -0,0 +1,39 @@
+﻿using Aplib.Core;
+using Aplib.Core.Desire.Goals;
+using System;
```
Comment (SilasPeters) [inline Aplib.Integrations.Unity/TestRunner.cs:3]: These are unused.

File: Aplib.Integrations.Unity.Tests/TestRunnerTests.cs (hunk 3 above)
Comment (SilasPeters) [inline Aplib.Integrations.Unity.Tests/TestRunnerTests.cs:10]: No xml docs :eye: :eye: 

How did SonarCloud pass?

**Review by SilasPeters — CHANGES_REQUESTED** [review]

**Comment by joachimdekker** [comment]
Thanks for commenting guys! I will not merge this PR, as we will move the aplib-unity integration to the Unity side.

**Comment by JensSteenmetz** [comment]
> Thanks for commenting guys! I will not merge this PR, as we will move the aplib-unity integration to the Unity side.

Sounds good! Please ping me when you're done!

**Comment by SilasPeters** [comment]
Why is this moved to unity?

I always envisioned a seperate repo for the unity extensions, or a different project. Not just moving the code to the unity demo

**Comment by joachimdekker** [comment]
We can do this in a seperate repo, but lets just make it work first for the demo. We can create a seperate `Aplib-Unity` repository which is a unity package.

**Pull request closed without merging**